src/commonMain/kotlin/fr/acinq/lightning/wire/LightningMessages.kt

```diff
@@ -1747,6 +1751,50 @@ data class PhoenixAndroidLegacyInfo(
     }
 }
 
+data class DNSAddressRequest(val offer: OfferTypes.Offer, val languageSubtype: String) : LightningMessage {
+
+    override val type: Long get() = DNSAddressRequest.type
+
+    override fun write(out: Output) {
+        val serializedOffer = offer.encode()
+        LightningCodecs.writeU16(serializedOffer.length, out)
+        LightningCodecs.writeBytes(serializedOffer.toByteArray(charset = Charsets.UTF_8), out)
```

Comment:
It's a bit weird to use a string encoding and then convert to bytes when we actually have a byte encoding for offers (and `encode` internally first encodes to bytes before going to a string representation using bech32m)...an offer is just a TLV stream, and we have the `Offer.tlvSerializer` available to directly get bytes?

Reply:
Great point, done in ad3b2efa.